File: Programmers/Lv3/Lv3_거스름돈.cpp
#include <string>
#include <vector>

using namespace std;

// 그냥 BFS는 시간초과
// DP를 사용하자!
int solution99(int n, vector<int> money) {
	int MOD = 1000000007;
	int answer = 0;
	vector<int> d(n + 1, 0);
	d[0] = 1;

	for (int i = 0; i < money.size(); i++) {
		for (int j = money[i]; j <= n ; j++) {
			d[j] = d[j] + d[j - money[i]] % MOD;
		}
	}

	return d[n];
}

File: Programmers/Lv3/Lv3_거스름돈.py
def solution(n, money):
    d = [0]*(n+1)
    d[0] = 1

    for i in money:
        for j in range(i,n+1):
            d[j] = (d[j] + d[j-i])%1000000007
    return d[n]

print(solution(5,[1,2,5]))